tests/src/FunctionalJavascript/MessageFormTest.php

```diff
@@ -133,6 +137,9 @@ public function testCorporateForm(): void {
     $this->assertTrue(strpos($captured_emails[1]['body'], 'Test subject') !== FALSE);
     $this->assertTrue(strpos($captured_emails[1]['body'], 'Test message') !== FALSE);
     $this->assertTrue(strpos($captured_emails[1]['body'], $topics['0']['topic_name']) !== FALSE);
+
+    // Assert that there was no redirection.
```

Comment:
Comment is misleading: in fact there is a redirection but one to the current page. Reading the rest of the tests doesn't indicate that normal behaviour would be to send the user to the homepage. So let's just say:

`Assert that instead of the user being redirected to the homepage, they are redirected to the same, contact form page`.

Also, you are missing coverage for it being a corporate form and having set a redirect path: `if (!$contact_form->getRedirectPath()) {`.

Reply:
its terminology, if the destination is same page then its a reload not a redirect

Reply:
@drishu Yeah, well, your code is performing a redirect, this is a functional (technical test).